app/controllers/reviews_controller.rb

```diff
@@ -36,10 +36,12 @@ def review_params
   end
 
   def set_book
-    @book = Book.find(params[:book_id])
+    @book = Rails.cache.fetch("books/#{params[:book_id]}") do
+      Book.find(params[:book_id])
+    end
```

Comment:
Normally, we cache only `index` or `show`. For update, we should query from db for updating for correctness of data krub